libkbfs/file_data.go

```diff
@@ -492,17 +539,22 @@ func (fd *fileData) getByteSlicesInOffsetRange(ctx context.Context,
 	// Grab the relevant byte slices from each block described by the
 	// indirect pointer, filling in holes as needed.
 	var bytes [][]byte
-	for _, iptr := range iptrs {
+	for i, iptr := range iptrs {
 		block := blockMap[iptr.BlockPointer]
 		blockLen := int64(len(block.Contents))
 		nextByte := nRead + startOff
 		toRead := n - nRead
 		blockOff := iptr.Off
 		lastByteInBlock := blockOff + blockLen
 
+		nextIPtrOff := nextBlockOff
+		if i < len(iptrs)-1 {
+			nextIPtrOff = iptrs[i+1].Off
+		}
+
 		if nextByte >= lastByteInBlock {
-			if nextBlockOff > 0 {
-				fill := nextBlockOff - nextByte
+			if nextIPtrOff > 0 {
```

Comment:
This was a somewhat unrelated bug when there was a hole in the middle of a file, followed by one or more blocks, and a read spanned both the hole and the blocks all the way to the end of the file.